src/main/java/core/basesyntax/serviceimpl/CsvToFruitTransactionConverter.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CsvToFruitTransactionConverter implements DataConverter<List<String>,
+        List<FruitTransaction>> {
+    private static final int TYPE = 0;
+    private static final int FRUIT = 1;
+    private static final int QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(line -> {
+                    String[] parts = line.split(",");
+                    return new FruitTransaction(
+                            FruitTransaction.Operation.fromCode(parts[TYPE]),
+                            parts[FRUIT],
+                            Integer.parseInt(parts[QUANTITY])
+                    );
+                })
+                .collect(Collectors.toList());
+    }
```

Comment:
The method `convertToTransaction` should handle possible data format issues. Currently, if the input data is malformed (e.g., incorrect number of fields, invalid integers, etc.), the method will throw an unhandled exception. Consider adding validation for the input data and throwing a custom, informative exception if the data is invalid.